### ISSUE TITLE
train_interactions w/ disjoint sets

### ISSUE THREAD

**Issue by eggie5** [lyst/lightfm#326]
I don't understand what the `train_interactions` flag does. My understanding was that given, a non-disjoint train and test set it would not predict & rank the intersecting interactions. 

However, I have disjoint sets, `check_intersections` warning does not appear, and I get different evaluation results. 

If the two sets are disjoint, what is the routine doing?

**Comment by maciejkula** [comment]
To evaluation functions (as per [this line](https://github.com/lyst/lightfm/blob/master/lightfm/evaluation.py#L165))?

It sets the score of interactions that are in the train set to a very low value, to simulate the fact that in production systems it's common to avoid recommending items that the user has already consumed.

What do you mean when you say that 'you get different evaluation results'? Different from what? Different in what case?

**Comment by eggie5** [comment]
I have a disjoint set. If I run precision@k for w/ and w/o the `train_interactions` flag, I get different results. Does this make sense? 

I didn't think I would get different results, b/c I thought `train_interactions` would simulate the effect of a disjoint set -- the condition that I already have.

**Comment by maciejkula** [comment]
It does not. It simply sets the scores of items present in the training set to `MIN_FLOAT`.
